packages/workbox-webpack-plugin/src/lib/get-manifest-entries-from-compilation.js

```diff
@@ -41,21 +41,54 @@ function checkConditions(asset, compilation, conditions = []) {
 }
 
 /**
- * Returns the names of all the assets in a chunk group.
+ * Returns the names of all the assets in all the chunks in a chunk group,
+ * if provided a chunk group name.
+ * Otherwise, if provided a chunk name, return all the assets in that chunk.
+ * Otherwise, if there isn't a chunk group or chunk with that name, return null.
  *
- * @param {ChunkGroup} chunkGroup
+ * @param {Compilation} compilation
+ * @param {string} chunkOrGroup
+ * @return {Array<Asset>|null}
+ * @private
+ */
+function getNamesOfAssetsInChunkOrGroup(compilation, chunkOrGroup) {
+  const chunkGroup = compilation.namedChunkGroups &&
+      compilation.namedChunkGroups.get(chunkOrGroup);
+  if (chunkGroup) {
+    const assetNames = [];
+    for (const chunk of chunkGroup.chunks) {
+      assetNames.push(...getNamesOfAssetsInChunk(chunk));
+    }
+    return assetNames;
+  } else {
+    const chunk = compilation.namedChunks &&
+        compilation.namedChunks.get(chunkOrGroup);
+    if (chunk) {
+      return getNamesOfAssetsInChunk(chunk);
+    }
+  }
+
+  // If we get here, there's no chunkGroup or chunk with that name.
+  return null;
+}
+
+/**
+ * Returns the names of all the assets in a chunk.
+ *
+ * @param {Chunk} chunk
  * @return {Array<Asset>}
  * @private
  */
-function getNamesOfAssetsInChunkGroup(chunkGroup) {
+function getNamesOfAssetsInChunk(chunk) {
   const assetNames = [];
-  for (const chunk of chunkGroup.chunks) {
-    assetNames.splice(0, 0, ...chunk.files);
-    // This only appears to be set in webpack v5.
-    if (chunk.auxiliaryFiles) {
-      assetNames.splice(0, 0, ...chunk.auxiliaryFiles);
-    }
+
+  assetNames.push(...chunk.files);
```

Comment:
I... honestly can't remember why this was written as a `splice(0, 0, ...)` previously. I don't believe that it matters whether the items are prepended or appended.

Reply:
If the ordering of these lists ends up determining the order of the URLs in the cache manifest, maybe this was done to prioritize resources? Prepending seems arbitrary though, I can't think of any reason to prioritize assets from chunk groups over assets from un-grouped chunks.

Reply:
Nah, we do an explicit sort on the `url` property of each entry later on. I think it was just a mistaken optimization on my part.